.github/workflows/sphinx.yml

```diff
@@ -0,0 +1,120 @@
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.8
+
+      - name: Cache pip
+        uses: actions/cache@v2
+        with:
+          path: ~/.cache/pip
+          key: ${{ runner.os }}-pip-${{ hashFiles('setup.py') }}
+          restore-keys: |
+            ${{ runner.os }}-pip-
+            ${{ runner.os }}-
+
+      - name: Install dependencies
+        run: |
+          cd stylist
+          python -m pip install --upgrade pip
+          pip install .[docs]
+
+      - name: Debugging information
+        run: |
+          echo "github.ref:" ${{github.ref}}
+          echo "github.event_name:" ${{github.event_name}}
+          echo "github.head_ref:" ${{github.head_ref}}
+          echo "github.base_ref:" ${{github.base_ref}}
+          set -x
+          cd stylist
+          git rev-parse --abbrev-ref HEAD
+          git branch
+          git branch -a
+          git remote -v
+          python -V
+          pip list --not-required
+          pip list
+
+      - name: Build Sphinx docs
+        run: |
+          cd stylist/documentation
+          make html
+
+      # Publish
+      #
+      - name: Copy new build (trunk)
+        if: |
+          contains(github.event_name, 'push')
+          && contains(github.ref, env.TRUNK_BRANCH)
+        run: |
+          set -x
+          cd gh-pages
+          find . -mindepth 1 ! -path 'branch*' -exec git rm {} \;
+          rsync -a ../stylist/documentation/build/html/ gh-pages/
+          find . -mindepth 1 ! -path 'branch*' -exec git add {} \;
+
+      - name: Copy new build (branch)
+        if: |
+          contains(github.event_name, 'push')
+          && !contains(github.ref, env.TRUNK_BRANCH)
+        run: |
+          set -x
+          brname="${{github.ref}}"
+          brname="${brname##refs/heads/}"
+          brdir=${brname//\//--}  # replace '/' with '--'
+          cd gh-pages
+          git rm -r branch/${brdir}
+          mkdir -p branch
+          rsync -a ../stylist/documentation/build/html branch/${brdir}
+          git add branch/${brdir}
+
+      - name: Delete old feature branches
+        if: ${{ github.event_name == 'push' }}
+        run: |
+          set -x
+          for brdir in $(ls gh-pages/branch/) ; do
+              brname=${brdir//--//\/}  # replace '--' with '/'
+              if ! git -C stylist show-ref remotes/origin/$brname ; then
+                  echo "Removing $brdir"
+                  git -C gh-pages rm -r branch/$brdir
+              fi
+          done
+
+      - name: Add "no jekyll" file
+        if: ${{ github.event_name == 'push' }}
+        run: |
+          touch gh-pages/.nojekyll
+
+      - name: Deploy
+        if: ${{ github.event_name == 'push' }}
+        run: |
+          set -x
+          git_hash=$(git -C stylist rev-parse --short "${{ github.sha }}")
+          git -C gh-pages -c "user.name=Sphinx Workflow" -c user.email=sphinx@example.com commit -a -m "Documentation from $git_hash"
```

Comment:
What are the purposes of the `-c` options?

Reply:
It causes `git` to change to that directory before acting. It means you don't need to `cd <working copy>` in the script.